### PR TITLE
LPS-138444 ObjectEntry search

### PR DESCRIPTION
**Pull request by BryanEngler**
let me know if you have any questions

https://issues.liferay.com/browse/LPS-138444

### CODE REVIEW

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/index/contributor/ObjectEntryModelDocumentContributor.java [hunk 1]
```diff
@@ -95,6 +110,27 @@ private void _contribute(Document document, ObjectEntry objectEntry)
 		document.addKeyword(
 			"objectDefinitionId", objectEntry.getObjectDefinitionId());
 
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				objectEntry.getObjectDefinitionId());
+
+		document.addKeyword(
+			"objectDefinitionName", objectDefinition.getShortName());
+
```
Comment (BryanEngler) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/index/contributor/ObjectEntryModelDocumentContributor.java]: this is the section of code related to the "title" fields

**Comment by BryanEngler** [comment]
i previously ran CI against this branch (https://github.com/BryanEngler/liferay-portal/pull/614#issuecomment-921148880) and noticed there were some failures in `relevant`, but it looks like those were also occurring on upstream so i think everything else looked good

**Comment by BryanEngler** [comment]
also just wanted to ask if you guys had any thought about what should happen when a user clicks on a search result. is there a page you guys have that lets you "view" an object entry?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/index/contributor/ObjectEntryModelDocumentContributor.java (hunk 1 above)
Comment (guilhermedcamacho) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/index/contributor/ObjectEntryModelDocumentContributor.java]: @marco-leo is working on create a configuration on Object Definition admin UI allowing user to select which Object Field represents the title of it

**Comment by guilhermedcamacho** [comment]
> also just wanted to ask if you guys had any thought about what should happen when a user clicks on a search result. is there a page you guys have that lets you "view" an object entry?

Yes, we have an auto generated portlet to see the entries, so the user is able to select one entry to view the details of it in another screen. I think you want to use this detail screen.

In Object's admin UI, when you are creating a new Object Definition there is a field "Panel Category Key" where you can choose where this portlet will be available, select it and then hit the save button, after that you can publish the Object and check this screen.

If you have any doubts, please reach the team on #t-objects

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/index/contributor/ObjectEntryModelDocumentContributor.java [hunk 2]
```diff
@@ -95,6 +110,27 @@ private void _contribute(Document document, ObjectEntry objectEntry)
...
+		//TODO
+		//String titleFieldPrefix = objectDefinition.getTitleFieldPrefix();
```
Comment (BryanEngler) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/index/contributor/ObjectEntryModelDocumentContributor.java]: @marco-leo see https://liferay.slack.com/archives/C02DXTT06LQ/p1631221892023100?thread_ts=1631221577.022400&cid=C02DXTT06LQ

**Comment by BryanEngler** [comment]
ci:tes:sf

**Comment by BryanEngler** [comment]
ci:test:sf

**Pull request closed without merging**